python/pyarrow/gandiva.pyx

```diff
@@ -150,10 +163,49 @@ cdef class Filter:
             batch.sp_batch.get()[0], selection))
         return SelectionVector.create(selection)
 
+
+cdef inline DataType _ensure_type(object type):
+    if type is None:
+        return None
+    elif not isinstance(type, DataType):
+        return type_for_alias(type)
+    else:
+        return type
+
+
 cdef class TreeExprBuilder:
 
-    def make_literal(self, value):
-        cdef shared_ptr[CNode] r = TreeExprBuilder_MakeLiteral(value)
+    def make_literal(self, value, dtype):
+        cdef shared_ptr[CNode] r
+        cdef DataType type = _ensure_type(dtype)
```

Comment:
Tests for passing types as string and a failing case?

Reply:
Added, thanks!

Reply:
Thanks!